File: DotNet/CesiumLanguageWriter/Advanced/CesiumRgbaValuePropertyAdaptor.cs
using System.Drawing;
using JetBrains.Annotations;

namespace CesiumLanguageWriter.Advanced
{
    /// <summary>
    /// Adapts a class that implements <see cref="ICesiumRgbaValuePropertyWriter"/> to implement
    /// <see cref="ICesiumInterpolatableValuePropertyWriter{TValue}"/> for <see cref="Color"/> values.
    /// </summary>
    /// <typeparam name="TFrom">The class that implements <see cref="ICesiumRgbaValuePropertyWriter"/> to adapt.</typeparam>
    public class CesiumRgbaValuePropertyAdaptor<TFrom> : CesiumInterpolatableWriterAdaptor<TFrom, Color>
        where TFrom : class, ICesiumRgbaValuePropertyWriter
    {
        /// <summary>
        /// Initializes a new instance.
        /// </summary>
        /// <param name="parent">The instance to wrap.</param>
        /// <param name="writeValueCallback">The callback to write values of type <see cref="Color"/>.</param>
        /// <param name="writeSamplesCallback">The callback to write samples of type <see cref="Color"/>.</param>
        /// <param name="writeDeleteValueCallback">The callback to write an indication that the client should delete existing data.</param>
        public CesiumRgbaValuePropertyAdaptor([NotNull] TFrom parent,
                                              [NotNull] CesiumWriterAdaptorWriteCallback<TFrom, Color> writeValueCallback,
                                              [NotNull] CesiumWriterAdaptorWriteSamplesCallback<TFrom, Color> writeSamplesCallback,
                                              [NotNull] CesiumWriterAdaptorWriteDeleteCallback<TFrom> writeDeleteValueCallback)
            : base(parent, writeValueCallback, writeSamplesCallback, writeDeleteValueCallback)
        {
        }
    }
}

File: Java/CesiumLanguageWriter/translatedSrc/cesiumlanguagewriter/advanced/CesiumRgbaValuePropertyAdaptor.java
package cesiumlanguagewriter.advanced;


import agi.foundation.compatibility.*;
import cesiumlanguagewriter.*;
import java.awt.Color;
import javax.annotation.Nonnull;

/**
 *  
 Adapts a class that implements {@link ICesiumRgbaValuePropertyWriter} to implement
 {@link ICesiumInterpolatableValuePropertyWriter} for {@link Color} values.
 
 

 * @param <TFrom> The class that implements {@link ICesiumRgbaValuePropertyWriter} to adapt.
 */
@SuppressWarnings( {
        "unused",
        "deprecation",
        "serial"
})
public class CesiumRgbaValuePropertyAdaptor<TFrom extends ICesiumRgbaValuePropertyWriter> extends CesiumInterpolatableWriterAdaptor<TFrom, Color> {
    /**
    *  
    Initializes a new instance.
    
    
    
    
    

    * @param parent The instance to wrap.
    * @param writeValueCallback The callback to write values of type {@link Color}.
    * @param writeSamplesCallback The callback to write samples of type {@link Color}.
    * @param writeDeleteValueCallback The callback to write an indication that the client should delete existing data.
    */
    public CesiumRgbaValuePropertyAdaptor(@Nonnull TFrom parent, @Nonnull CesiumWriterAdaptorWriteCallback<TFrom, Color> writeValueCallback,
            @Nonnull CesiumWriterAdaptorWriteSamplesCallback<TFrom, Color> writeSamplesCallback, @Nonnull CesiumWriterAdaptorWriteDeleteCallback<TFrom> writeDeleteValueCallback) {
        super(parent, writeValueCallback, writeSamplesCallback, writeDeleteValueCallback);
    }
}